lib/completions/endpoints/hugging_face.rb

```diff
@@ -30,9 +28,13 @@ def prepare_payload(prompt, model_params)
             .tap do |payload|
               payload[:parameters].merge!(model_params)
 
-              token_limit = 2_000 || SiteSetting.ai_hugging_face_token_limit
+              token_limit = SiteSetting.ai_hugging_face_token_limit || 4_000
 
               payload[:parameters][:max_new_tokens] = token_limit - prompt_size(prompt)
+
+              if @streaming_mode
```

Comment:
Moving to headers instead of path makes it compatible with both self-hosting and hosted API